sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/mdesupport/EncryptionKeyStoreProviderImpl.java

```diff
@@ -0,0 +1,98 @@
+import com.microsoft.data.encryption.cryptography.MicrosoftDataEncryptionException;
+
+
+class EncryptionKeyStoreProviderImpl extends EncryptionKeyStoreProvider {
+    private final EncryptionKeyWrapProvider encryptionKeyWrapProvider;
+    public EncryptionKeyStoreProviderImpl(EncryptionKeyWrapProvider encryptionKeyWrapProvider) {
+        this.encryptionKeyWrapProvider = encryptionKeyWrapProvider;
+    }
+
+    /**
+     * Getter for provider name.
+     *
+     * @return provider name
+     */
+    @Override
+    public String getProviderName() {
+        return this.encryptionKeyWrapProvider.getProviderName();
+    }
+
+    /**
+     * Unwraps the specified encryptedKey of a data encryption key. The encrypted value is expected to be encrypted
+     * using the key encryption key with the specified encryptionKeyId and using the specified algorithm.
+     *
+     * @param encryptionKeyId
+     *        The key Id tells the provider where to find the key.
+     * @param algorithm
+     *        The encryption algorithm.
+     * @param encryptedKey
+     *        The ciphertext key.
+     * @return The unwrapped data encryption key.
+     * @throws MicrosoftDataEncryptionException
+     *         on error
+     */
+    @Override
+    public byte[] unwrapKey(String encryptionKeyId, KeyEncryptionKeyAlgorithm algorithm, byte[] encryptedKey) throws MicrosoftDataEncryptionException {
+        return this.encryptionKeyWrapProvider.unwrapKeyAsync(encryptionKeyId, algorithm.toString(), encryptedKey);
+    }
+
+    /**
+     * Wraps a data encryption key using the key encryption key with the specified encryptionKeyId and using the
+     * specified algorithm.
+     *
+     * @param encryptionKeyId
+     *        The key Id tells the provider where to find the key.
+     * @param algorithm
+     *        The encryption algorithm.
+     * @param key
+     *        The plaintext key
+     * @return The wrapped data encryption key.
+     * @throws MicrosoftDataEncryptionException
+     *         on error
+     */
+    @Override
+    public byte[] wrapKey(String encryptionKeyId, KeyEncryptionKeyAlgorithm algorithm, byte[] key) throws MicrosoftDataEncryptionException {
+        return this.encryptionKeyWrapProvider.wrapKeyAsync(encryptionKeyId, algorithm.toString(), key);
+    }
+
+    /**
+     *  The public facing Cosmos Encryption library interface does not expose this method, hence not supported.
+     *
+     * @param encryptionKeyId
+     *        The key Id tells the provider where to find the key.
+     * @param allowEnclaveComputations
+     *        Indicates whether the key encryption key supports enclave computations.
+     * @return The signature of the key encryption key metadata.
+     * @throws MicrosoftDataEncryptionException
+     *         on error
+     */
+    @Override
+    public byte[] sign(String encryptionKeyId, boolean allowEnclaveComputations) {
+        return new byte[0]; // "The Sign operation is not supported. " ;
```

Comment:
if not supported, should we throw exception instead?

Reply:
+1 here

Reply:
That's what i started with but AAP KeyEncryptionKey.java constructor calling this method , if this would have been async/lazy like .net it would have make sense to throw whenever called as this is not getting called after KeyEncryptionKey initialization.
But in Java we have two choice either return null and empty array (no signing) , so i choose empty array , verified with .NET encryption  dev